azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -124,12 +127,46 @@ def add_term(self, c: Union[int, float], indices: List[int]):
         self.terms.append(Term(indices=indices, c=c))
         self.uploaded_blob_uri = None
 
-    def add_terms(self, terms: List[Term]):
-        """Adds a list of terms to the `Problem` representation
+    def add_terms(self, terms: List[Term],
+                  type: str = None, c: Union[int, float] = 1):
+        """Adds an optionally grouped list of monomial terms 
+        to the `Problem` representation
 
         :param terms: The list of terms to add to the problem
+        :param type: Type of grouped term being added, if applicable
+        :param c: Weight of grouped term, if applicable
+        """
+        if type:
+            # Grouped term
+            if type == 'na':
+                gtype = GroupType.combination
+            elif type == 'slc':
+                gtype = GroupType.squared_linear_combination
+            else:
```

Comment:
wont be needed if you use enum for the type

Reply:
When type is entered as the GroupType, the string comparisons are now skipped over.

Reply:
Why allow for strings in the first place when there's an enum available?